src/adapter/src/catalog/storage.rs

```diff
@@ -555,29 +556,46 @@ impl<S: Append> Connection<S> {
 
         // Initialize connection.
         initialize_stash(&mut stash).await?;
+
+        // Choose a time at which to boot. This is the time at which we will run
+        // internal migrations, and is also exposed upwards in case higher
+        // layers want to run their own migrations at the same timestamp.
+        //
+        // This time is usually the current system time, but with protection
+        // against backwards time jumps, even across restarts.
+        let previous_now_ts = try_get_persisted_timestamp(&mut stash, &Timeline::EpochMilliseconds)
+            .await?
+            .unwrap_or(mz_repr::Timestamp::MIN);
```

Comment:
When will this ever be MIN? If we initialize the stash with it it should always exist.

Reply:
It's `None` on initial bootstrapping. `initialize_stash` above just creates the collections as empty. Previously the initial value for this timeline would be installed in the first migration, but this code has to run _before_ the first migration. So I removed that bit of the migration. This is now the one place where the stash initialization code reads/writes the persistedtimestamp.

Reply:
Ah I missed the call below to monotonic now.